os/compression/compress_read.c

```diff
@@ -108,31 +108,47 @@ static void compress_blocks_to_read(int *first_block, int *last_block, int *no_b
  *
  * Returned value:
  *   OK (0) is Success
- *   ERROR (-1) on Failure
+ *   Negative value on Failure
  ****************************************************************************/
 static int compress_parse_header(int filfd, uint16_t offset)
 {
 	off_t rpos;					/* Position returned by lseek */
 	int nbytes;					/* Number of bytes read  */
+	int compheader_size;				/* Total size of compression header */
 
-	/* Seek to the next read position */
-
+	/* Seek to location of size of compression header */
 	rpos = lseek(filfd, offset, SEEK_SET);
 	if (rpos != offset) {
 		int errval = get_errno();
-		bmdbg("ERROR : leek to offset %lu failed: %d\n", (unsigned long)offset, errval);
+		bmdbg("ERROR : lseek to offset %lu failed: %d\n", (unsigned long)offset, errval);
 		return -errval;
 	}
 
-	/* Read compression header data from the file data from offset */
+	/* Read compression header size from the file data */
+	nbytes = read(filfd, &compheader_size, sizeof(compheader_size));
+	if (nbytes != sizeof(compheader_size)) {
+		bmdbg("Read for compression header size from offset %lu failed\n", offset);
+		return ERROR;
+	}
+
+	/* Allocate memory for compression header now that we know it's size */
+	compression_header = (struct s_header *)kmm_malloc(compheader_size);
+	if (!compression_header) {
+		berr("Failed kmm_malloc for compression_header\n");
```

Comment:
Why do you use `berr` instead of bmdbg?

Reply:
Done. Made this change.
`berr` -> `bmdbg`